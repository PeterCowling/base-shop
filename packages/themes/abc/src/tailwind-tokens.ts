// packages/themes/abc/tailwind-tokens.ts
//
// Default design token overrides for the ABC theme. These values can be
// merged with the base theme tokens to style a shop.

export const tokens = {
  "--color-bg": "0 0% 100%",
  "--color-fg": "0 0% 10%",
  "--color-primary": "160 80% 40%",
  "--color-primary-fg": "0 0% 100%",
  "--color-accent": "200 90% 45%",
  "--color-success": "142 76% 97%",
  "--color-success-fg": "142 72% 30%",
  "--color-warning": "40 90% 96%",
  "--color-warning-fg": "25 85% 31%",
  "--color-info": "210 90% 96%",
  "--color-info-fg": "210 90% 35%",
  "--color-muted": "0 0% 88%",
  "--font-sans": 'var(--font-geist-sans)',
  "--font-mono": 'var(--font-geist-mono)',
  "--space-1": "4px",
  "--space-2": "8px",
  "--space-3": "12px",
  "--space-4": "16px",
  "--radius-sm": "4px",
  "--radius-md": "8px",
  "--radius-lg": "12px",
  "--shadow-sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--shadow-md":
    "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
  "--shadow-lg":
    "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
} as const;
