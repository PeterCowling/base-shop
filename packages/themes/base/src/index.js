// packages/themes/base/src/index.ts
export * from "./tokens";
