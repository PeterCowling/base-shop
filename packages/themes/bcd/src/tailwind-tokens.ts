// packages/themes/bcd/tailwind-tokens.ts
//
// Default design token overrides for the BCD theme.

export const tokens = {
  "--color-bg": "0 0% 100%",
  "--color-fg": "0 0% 10%",
  "--color-primary": "340 80% 50%",
  "--color-primary-fg": "0 0% 100%",
  "--color-accent": "40 95% 50%",
  // hero gradient stops (indigo → purple → slate)
  "--gradient-hero-from": "234 89% 60%",
  "--gradient-hero-via": "270 83% 60%",
  "--gradient-hero-to": "222 47% 11%",
  "--color-success": "142 76% 97%",
  "--color-success-fg": "142 72% 30%",
  "--color-warning": "40 90% 96%",
  "--color-warning-fg": "25 85% 31%",
  "--color-info": "210 90% 96%",
  "--color-info-fg": "210 90% 35%",
  "--color-muted": "0 0% 88%",
  "--font-sans": 'var(--font-geist-sans)',
  "--font-mono": 'var(--font-geist-mono)',
  "--space-1": "4px",
  "--space-2": "8px",
  "--space-3": "12px",
  "--space-4": "16px",
  "--radius-sm": "4px",
  "--radius-md": "8px",
  "--radius-lg": "12px",
  "--shadow-sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--shadow-md":
    "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
  "--shadow-lg":
    "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
} as const;
