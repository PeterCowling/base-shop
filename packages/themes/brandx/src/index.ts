export const tokens = {
  "--color-bg": { light: "purple" },
};
