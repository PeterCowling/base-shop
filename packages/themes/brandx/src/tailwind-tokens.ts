// packages/themes/brandx/tailwind-tokens.ts
//
// Default design token overrides for the brandx theme.

export const tokens = {
  "--color-bg": "0 0% 100%",
  "--color-bg-dark": "0 0% 4%",
  "--color-fg": "0 0% 10%",
  "--color-fg-dark": "0 0% 93%",
  "--color-primary": "340 80% 50%",
  "--color-primary-dark": "340 80% 60%",
  "--color-primary-fg": "0 0% 100%",
  "--color-primary-fg-dark": "0 0% 10%",
  "--color-accent": "40 95% 50%",
  "--color-accent-dark": "40 95% 60%",
  "--color-success": "142 76% 97%",
  "--color-success-dark": "142 72% 27%",
  "--color-success-fg": "142 72% 30%",
  "--color-success-fg-dark": "142 70% 94%",
  "--color-warning": "40 90% 96%",
  "--color-warning-dark": "35 90% 30%",
  "--color-warning-fg": "25 85% 31%",
  "--color-warning-fg-dark": "40 90% 96%",
  "--color-info": "210 90% 96%",
  "--color-info-dark": "210 90% 35%",
  "--color-info-fg": "210 90% 35%",
  "--color-info-fg-dark": "210 90% 96%",
  "--color-muted": "0 0% 88%",
  "--color-muted-dark": "0 0% 60%",
  "--font-sans": 'var(--font-geist-sans)',
  "--font-mono": 'var(--font-geist-mono)',
  "--space-1": "4px",
  "--space-2": "8px",
  "--space-3": "12px",
  "--space-4": "16px",
  "--radius-sm": "4px",
  "--radius-md": "8px",
  "--radius-lg": "12px",
  "--shadow-sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--shadow-md":
    "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
  "--shadow-lg":
    "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
} as const;
