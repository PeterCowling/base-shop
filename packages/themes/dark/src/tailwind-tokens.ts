// packages/themes/dark/tailwind-tokens.ts
//
// Default design token overrides for the dark theme.

export const tokens = {
  "--color-bg": "0 0% 4%",
  "--color-fg": "0 0% 93%",
  "--color-primary": "220 90% 66%",
  "--color-primary-fg": "0 0% 10%",
  "--color-accent": "260 83% 15%",
  "--color-accent-fg": "0 0% 90%",
  "--color-danger": "0 63% 31%",
  "--color-danger-fg": "0 93% 94%",
  "--color-success": "142 72% 27%",
  "--color-success-fg": "142 70% 94%",
  "--color-warning": "35 90% 30%",
  "--color-warning-fg": "40 90% 96%",
  "--color-info": "210 90% 35%",
  "--color-info-fg": "210 90% 96%",
  "--color-muted": "0 0% 60%",
  "--font-sans": '"Geist Sans", system-ui, sans-serif',
  "--font-mono": '"Geist Mono", ui-monospace, monospace',
  "--space-1": "4px",
  "--space-2": "8px",
  "--space-3": "12px",
  "--space-4": "16px",
  "--radius-sm": "4px",
  "--radius-md": "8px",
  "--radius-lg": "12px",
  "--shadow-sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--shadow-md":
    "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
  "--shadow-lg":
    "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
} as const;
