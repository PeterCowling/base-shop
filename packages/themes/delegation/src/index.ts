export const tokens = { '--color-bg': { light: '#def' } };