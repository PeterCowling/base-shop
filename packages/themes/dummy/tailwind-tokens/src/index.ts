export const tokens = { "--color-bg": "green" };
