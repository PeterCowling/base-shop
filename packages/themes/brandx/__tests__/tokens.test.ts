import fs from 'node:fs';
import path from 'node:path';
import { tokens } from '../src/tailwind-tokens';

const expectedKeys = [
  '--color-bg',
  '--color-fg',
  '--color-primary',
  '--color-primary-fg',
  '--color-accent',
  '--color-muted',
  '--font-sans',
  '--font-mono',
  '--space-1',
  '--space-2',
  '--space-3',
  '--space-4',
  '--radius-sm',
  '--radius-md',
  '--radius-lg',
  '--shadow-sm',
  '--shadow-md',
  '--shadow-lg',
] as const;

describe('brandx theme tokens', () => {
  test('exports expected token keys', () => {
    expect(Object.keys(tokens).sort()).toEqual([...expectedKeys].sort());
  });

  test('css contains token variables', () => {
    const cssPath = path.join(__dirname, '..', 'src', 'tokens.css');
    const css = fs.readFileSync(cssPath, 'utf8');
    for (const key of expectedKeys) {
      expect(css).toContain(key);
    }
  });

  test('tokens snapshot', () => {
    expect(tokens).toMatchSnapshot();
  });
});
