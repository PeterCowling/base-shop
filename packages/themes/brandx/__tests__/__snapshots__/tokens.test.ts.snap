// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brandx theme tokens tokens snapshot 1`] = `
{
  "--color-accent": "40 95% 50%",
  "--color-accent-dark": "40 95% 60%",
  "--color-bg": "0 0% 100%",
  "--color-bg-dark": "0 0% 4%",
  "--color-fg": "0 0% 10%",
  "--color-fg-dark": "0 0% 93%",
  "--color-info": "210 90% 96%",
  "--color-info-dark": "210 90% 35%",
  "--color-info-fg": "210 90% 35%",
  "--color-info-fg-dark": "210 90% 96%",
  "--color-muted": "0 0% 88%",
  "--color-muted-dark": "0 0% 60%",
  "--color-primary": "340 80% 50%",
  "--color-primary-dark": "340 80% 60%",
  "--color-primary-fg": "0 0% 100%",
  "--color-primary-fg-dark": "0 0% 10%",
  "--color-success": "142 76% 97%",
  "--color-success-dark": "142 72% 27%",
  "--color-success-fg": "142 72% 30%",
  "--color-success-fg-dark": "142 70% 94%",
  "--color-warning": "40 90% 96%",
  "--color-warning-dark": "35 90% 30%",
  "--color-warning-fg": "25 85% 31%",
  "--color-warning-fg-dark": "40 90% 96%",
  "--font-mono": ""Geist Mono", ui-monospace, monospace",
  "--font-sans": ""Geist Sans", system-ui, sans-serif",
  "--radius-lg": "12px",
  "--radius-md": "8px",
  "--radius-sm": "4px",
  "--shadow-lg": "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
  "--shadow-md": "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
  "--shadow-sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
  "--space-1": "4px",
  "--space-2": "8px",
  "--space-3": "12px",
  "--space-4": "16px",
}
`;
